db/column_family.cc

```diff
@@ -562,39 +562,20 @@ std::unique_ptr<WriteControllerToken> SetupDelay(
   return write_controller->GetDelayToken(write_rate);
 }
 
-int GetL0ThresholdSpeedupCompaction(int level0_file_num_compaction_trigger,
-                                    int level0_slowdown_writes_trigger) {
+int64_t GetL0ThresholdSpeedupCompaction(
+    int64_t level0_file_num_compaction_trigger,
```

Comment:
This bases on the condition that `int` is `int32_t` so that `int64_t` is longer and there is an implicit cast in the caller of GetL0ThresholdSpeedupCompaction() where the return value is compared to an int which is automatically casted to int64_t. Can we make this implicit solution more explicit? For example, we still use int when as parameter and return value, but inside the function, we explicitly cast it to long?

Or can we do something really simple: if either level0_file_num_compaction_trigger or level0_slowdown_writes_trigger is more than std::numeric_limits<int>::max() / 2, just return level0_file_num_compaction_trigger?